### PR TITLE
Add multiple priority timeslice test group

### PR DESCRIPTION
**Pull request by chinglee-iot**
Add multiple priority timeslice test group

Description
-----------
The test result with configNUMBER_OF_CORES = 16
```
# run each test case
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/build/bin/smp_multiple_priorities_timeslice_multiple_priorities_timeslice_utest /mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/build/bin/smp_multiple_priorities_timeslice_covg_multiple_priorities_timeslice_utest
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:122:test_timeslice_verification_tasks_equal_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:187:test_timeslice_verification_idle_core:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:253:test_timeslice_verification_different_priority_tasks:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:322:test_timeslice_verification_low_priority_tasks_rotate:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:409:test_priority_change_tasks_different_priority_raise_to_equal:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:500:test_priority_change_tasks_equal_priority_raise_to_high:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:589:test_priority_change_tasks_equal_priority_lower_ready_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:674:test_priority_change_tasks_equal_priority_lower_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:762:test_task_create_tasks_equal_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:842:test_task_create_tasks_lower_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:922:test_task_create_tasks_higher_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1029:test_task_delete_tasks_equal_priorities_delete_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1127:test_task_suspend_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1241:test_task_block_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1357:test_task_affinity_verification:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1448:test_task_affinity_set_affinity_running_task:PASS

-----------------------
16 Tests 0 Failures 0 Ignored
```

Test result with configNUMBER_OF_CORES = 4
```
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:122:test_timeslice_verification_tasks_equal_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:194:test_timeslice_verification_idle_core:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:260:test_timeslice_verification_different_priority_tasks:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:329:test_timeslice_verification_low_priority_tasks_rotate:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:416:test_priority_change_tasks_different_priority_raise_to_equal:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:507:test_priority_change_tasks_equal_priority_raise_to_high:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:596:test_priority_change_tasks_equal_priority_lower_ready_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:681:test_priority_change_tasks_equal_priority_lower_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:769:test_task_create_tasks_equal_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:849:test_task_create_tasks_lower_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:929:test_task_create_tasks_higher_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1036:test_task_delete_tasks_equal_priorities_delete_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1134:test_task_suspend_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1248:test_task_block_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1364:test_task_affinity_verification:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1455:test_task_affinity_set_affinity_running_task:PASS

-----------------------
16 Tests 0 Failures 0 Ignored
```

Test result with configNUMBER_OF_CORES = 2
```
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:122:test_timeslice_verification_tasks_equal_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:187:test_timeslice_verification_idle_core:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:253:test_timeslice_verification_different_priority_tasks:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:322:test_timeslice_verification_low_priority_tasks_rotate:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:409:test_priority_change_tasks_different_priority_raise_to_equal:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:500:test_priority_change_tasks_equal_priority_raise_to_high:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:589:test_priority_change_tasks_equal_priority_lower_ready_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:674:test_priority_change_tasks_equal_priority_lower_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:762:test_task_create_tasks_equal_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:842:test_task_create_tasks_lower_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:922:test_task_create_tasks_higher_priority:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1029:test_task_delete_tasks_equal_priorities_delete_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1127:test_task_suspend_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1241:test_task_block_running_task:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1357:test_task_affinity_verification:PASS
/mnt/c/msys64/home/chinglee/smp/josh/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c:1448:test_task_affinity_set_affinity_running_task:PASS

-----------------------
16 Tests 0 Failures 0 Ignored
```

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by chinglee-iot** [comment]
Show the result with different core numbers.

File: FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c [hunk 1]
```diff
@@ -72,4 +76,1421 @@ int suiteTearDown( int numFailures )
...
+    verifySmpTask( &xTaskHandles[ i ], eReady, -1 );
+
+    /* Generate a tick for each configNUMBER_OF_CORES. This will cause each
+     * task to be either moved to the ready state or the running state */
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        xTaskIncrementTick_helper();
+
+        /* Verify the last created task runs on each core or enters the ready state */
+        verifySmpTask( &xTaskHandles[ configNUMBER_OF_CORES ], eRunning, i );
```
Comment (joshzarr) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c]: The comment is incorrect, you never verify the task enters the ready state.
Do you need to increment the tick one more time so the task enters the ready state?

File: FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c [hunk 2]
```diff
@@ -72,4 +76,1421 @@ int suiteTearDown( int numFailures )
...
+}
+
+/**
+ * @brief AWS_IoT-FreeRTOS_SMP_TC-91
+ * A high priority task will be created for each available CPU core. An
+ * additional low priority task will be created in the ready state. This test
+ * will verify that as OS ticks are generated all CPU cores will remain running
+ * their original tasks and the ready task never enters the running state.
+ *
+ * #define configRUN_MULTIPLE_PRIORITIES                    0
```
Comment (joshzarr) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c]: Multiple priorities 1

File: FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c [hunk 3]
```diff
@@ -72,4 +76,1421 @@ int suiteTearDown( int numFailures )
...
+    {
+        xTaskIncrementTick_helper();
+
+        for( j = 0; j < configNUMBER_OF_CORES; j++ )
+        {
+            verifySmpTask( &xTaskHandles[ j ], eRunning, j );
+        }
+    }
+
+    /* Create a new low priority task */
```
Comment (joshzarr) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c]: Typo, should be high priority task

**Review by joshzarr — COMMENTED** [review]
Code changes look good!
Verified tests pass when configured for 2/4/16 cores
Verified test cases match JAMA descriptions

Some minor comments/typos to be corrected prior to merging

File: FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c (hunk 1 above)
Comment (chinglee-iot) [reply FreeRTOS/Test/CMock/smp/multiple_priorities_timeslice/multiple_priorities_timeslice_utest.c]: Let's increment the tick one more time to verify the task enter ready state as well.

**Review by joshzarr — APPROVED** [review]

**Review by joshzarr — APPROVED** [review]

**Pull request merged**